Tools/SwiftBrowser/Source/BrowserViewModel.swift

```diff
@@ -39,17 +39,87 @@ extension PDF: Transferable {
     }
 }
 
+@MainActor
+final class DialogPresenter: DialogPresenting {
+    struct Dialog: Hashable, Identifiable, Sendable {
+        enum Configuration: Sendable {
+            case alert(String, @Sendable () -> Void)
+            case confirm(String, @Sendable (WebPage_v0.JavaScriptConfirmResult) -> Void)
+            case prompt(String, defaultText: String?, @Sendable (WebPage_v0.JavaScriptPromptResult) -> Void)
+        }
+
+        let id = UUID()
+        let configuration: Configuration
+
+        func hash(into hasher: inout Hasher) {
+            self.id.hash(into: &hasher)
+        }
+
+        static func == (lhs: Dialog, rhs: Dialog) -> Bool {
+            lhs.id == rhs.id
+        }
+    }
+
+    struct FilePicker {
+        let allowsMultipleSelection: Bool
+        let allowsDirectories: Bool
+        let completion: (WebPage_v0.FileInputPromptResult) -> Void
+    }
+
+    weak var owner: BrowserViewModel? = nil
+
+    func handleJavaScriptAlert(message: String, initiatedBy frame: WebPage_v0.FrameInfo) async {
+        await withCheckedContinuation { continuation in
+            owner?.currentDialog = Dialog(configuration: .alert(message) {
+                continuation.resume()
+            })
+        }
+    }
+
+    func handleJavaScriptConfirm(message: String, initiatedBy frame: WebPage_v0.FrameInfo) async -> WebPage_v0.JavaScriptConfirmResult {
+        await withCheckedContinuation { continuation in
+            owner?.currentDialog = Dialog(configuration: .confirm(message) {
```

Comment:
And likewise
```suggestion
            owner?.currentDialog = Dialog(configuration: .confirm(message, continuation.resume(returning:)))
```

Reply:
When I try this, I get

```
Converting a value of type '@Sendable (sending WebPage_v0.JavaScriptConfirmResult) -> ()' to type '@Sendable (WebPage_v0.JavaScriptConfirmResult) -> Void' risks causing data races
```

Reply:
I made the enum associated value have a `sending` parameter and that fixed it

Reply:
Hmm, I was expecting this to just convert, but some of the concurrency annotations are more strict (ex: you can generally pass a `(Int?) -> Void` to an `(Int) -> Void` because it's a subtype).